lib/automation.go

```diff
@@ -164,36 +163,28 @@ func (automationImpl) Deploy(scope scope, p *DeployParams) error {
 			return fmt.Errorf("deploy: given workflow and workflow on record have different OwnerIDs")
 		}
 	}
+	// Because deploy runs as a background task, re-root execution context atop
+	// the application context
+	scope = scope.ReplaceParentContext(scope.AppContext())
 	// TODO(ramfox): if we decide that you can interact with the automation subsystem when
 	// qri connect is NOT running, we need a `wait` flag in DeployParams, that, when `true`,
 	// does NOT deploy in a go routine
 	go deploy(scope, p)
 	return nil
 }
 
-func deploy(s scope, p *DeployParams) {
-	ctx := profile.AddIDToContext(s.AppContext(), s.ActiveProfile().ID.Encode())
-	// TODO(ramfox): we need the scope context to last longer than the context
-	// that was passed to us. We are, for now, going to rely on the app context
-	// in the future we will probably want something more sophisticated so that
-	// the user can cancel a deploy or run themselves
-	scope, err := newScope(ctx, s.inst, s.source)
-	if err != nil {
-		log.Debugw("deploy: creating new scope", "error", err)
-	}
+func deploy(scope scope, p *DeployParams) {
 	vi := dsref.ConvertDatasetToVersionInfo(p.Dataset)
 	ref := vi.SimpleRef().String()
 	deployPayload := event.DeployEvent{
 		Ref:        ref,
 		InitID:     p.Dataset.ID,
 		WorkflowID: p.Workflow.ID.String(),
 	}
-	log.Debugw("deploy started", "payload", deployPayload)
-	go func() {
-		if err := scope.Bus().PublishID(scope.Context(), event.ETAutomationDeployStart, ref, deployPayload); err != nil {
-			log.Debug(err)
-		}
-	}()
+
+	log.Debugw("deploy started", "ref", vi.SimpleRef().String(), "payload", deployPayload)
+	scope.sendEvent(event.ETAutomationDeployStart, ref, deployPayload)
```

Comment:
love the `sendEventAsync` function. I've moved it into a method of `scope` and dropped the required goroutine. IMHO go programs should use the `go` keyword to indicate async at the call site. I've also taken this opportunity to have some events _not_ fire async, which gives other subsystems a chance to fully handle the event before proceeding.